topless tattoo

picobyte opened the pull request.

aika092 commented on Trap Quest.inform/Inform/Extensions/Player/Exposure Addiction.i7x:
```diff
@@ -22,4 +22,26 @@ This is the knickers can't be summoned with no panties fetish rule:
+Definition: yourself is fighting against no-bra:
+	if topless permanent fetish is 1 or there is a worn topless tattoo or the humiliation of the player >= 35000:
+		let C be a random worn bra;
+		if C is bra and C is not no protection, decide yes;
```
`no protection` is a property of crotch covering, whereas we want to use `fully exposing` here (this is one of the values of the clothing-cleavage property).  So this line 30 should read

`		if C is bra and C is not fully exposing, decide yes;`

aika092 commented on Trap Quest.inform/Inform/Extensions/Player/Exposure Addiction.i7x:
```diff
@@ -22,4 +22,26 @@ This is the knickers can't be summoned with no panties fetish rule:
+	if the player is fighting against no-bra:
+		if braless reminder <= 0:
+			let C be a random worn bra;
+			if C is bra, say "[bold type]You [one of][or]still [stopping]feel terrible wearing your [printed name of C][bold type]!  [one of][if there is a worn topless tattoo]Your heart aches and you[else]You[end if] want to free your [BreastDesc]. It's lowers your awareness and consumes your thought.[or][stopping][roman type][line break]";
```
correct English would be "**It** lowers your awareness and consumes your **thoughts**".

aika092 commented on Trap Quest.inform/Inform/Extensions/Player/Exposure Addiction.i7x:
```diff
@@ -22,4 +22,26 @@ This is the knickers can't be summoned with no panties fetish rule:
+	if the player is fighting against no-bra, say "You feel very uncomfortable wearing this, [if there is a worn topless tattoo]the tattooed eye in your chest aches[else]because you know they want to see your [BreastDesc][end if].".
+This is the bra can't be summoned with braless fetish rule:
+	if summoning is 1 and (topless permanent fetish is 1 or the class of the player is faerie):
+		unless wearing-target is no protection, rule fails.
```
Again due to us looking at chest covering clothing and not crotch covering clothing, line 44 should read

`		unless wearing-target is fully exposing, rule fails.`

picobyte commented:
Thanks for your review. The ben wa balls patch compiles but gives an err on description. Can you still exclude it, or should I send a new pull request?

aika092 commented on Trap Quest.inform/Inform/Extensions/Sex Toys/Ben Wa Balls.i7x:
```diff
@@ -0,0 +1,43 @@
+	decide on figure of ben wa balls.
+
+To say ClothingDesc of (C - ben wa balls):
+	say "Two balls [if the item described is cursed and the curse-ID of the item described is sure]of plastic on a string. One pops out from time to time, embarassingly, but the other won't come out due to its curse[otherwise]made of metal[end if][if the item described is worn]. Their turning, when you move, slowly builds your arousal.[otherwise].[end if]".
```
You don't want to use `item described` here ideally, you want to use `C`. If you copied this from another item please tell me which one as that should be changed too. If ClothingDesc was ever called by any other means other than accessing the description of the item, it would behave incorrectly.

aika092 commented on Trap Quest.inform/Inform/Extensions/Player/Exposure Addiction.i7x:
```diff
@@ -0,0 +1,51 @@
+topless permanent fetish is a number that varies.  topless permanent fetish is 0.
+Definition: yourself is fighting against topless:
+	if topless permanent fetish is 1 or there is a worn topless tattoo or the humiliation of the player >= 35000:
+		let C be a random worn bra;
```
This is a pretty good implementation but personally I want to do it differently and also take into account some other things.

I would do it like this:

`Definition: yourself is fighting against topless:
    if topless permanent fetish is 1 or there is a worn topless tattoo or the humiliation of the player >= 35000:
        repeat with C running through worn breast covering clothing:
            if C is actually dense and C is top-in-place and C is not fully exposing and C is nipple covering, decide yes;
    decide no.`

I haven't tested that code to make sure all the terms are correct but you get the idea.  Also I think with the use of `nipple covering`, the `not fully exposing` becomes redundant.  But we can keep it in just in case.

Anyway, we now also need to change the "braless reminder" to make sure that it always warns the players that their breasts are covered.  We can do that with changing line 40 to `let C be a random worn actually dense top-in-place nipple covering clothing;` and the start of line 41 to `if C is clothing`.

I think it should also work to change line 44 to `Report wearing actually dense nipple covering clothing:`

aika092 commented on Trap Quest.inform/Inform/Extensions/Player/Exposure Addiction.i7x:
```diff
@@ -0,0 +1,51 @@
+	if the player is fighting against topless, say "You feel very uncomfortable wearing this, [if there is a worn topless tattoo]the tattooed eye in your chest aches[else]because you know they want to see your [BreastDesc][end if].".
+This is the bra can't be summoned with braless fetish rule:
+	if summoning is 1 and (topless permanent fetish is 1 or the class of the player is faerie):
+		unless wearing-target is fully exposing, rule fails.
```
I think we want to allow players to wear clothing that obeys the rules or can be top-displaced.  So I think we should change this to something like `unless wearing-target is fully exposing or wearing-target is see-through or wearing-target is not not-top-displacable or wearing-target is nipple exposing, rule fails.`

aika092 commented:
I'm very impressed with your regex skills!!! Thanks for making all those changes to the ClothingDesc functions.

picobyte commented:
Thanks, for particular required changes we could make github issues or a TODO file.

Yes I also found out as well the 5b54b3c patch was wrong. Seems like my testing was not working due to extensions stored outside the trapquest source directory. Was already wondering why I couldn't get the topless tattoo to work.

Of interest may also be the bed posts and skippy ball. And it looks like the bladder wasn't really working beyond a point.

picobyte commented:
The skippy ball was designed to have a few events, affected by glue, zeroG, poletraps and by the aeromancer, but somehow hooks don't seem to trigger. I can purloin the skippy, drop it and do the same with the aeromancer or glue, but glue-stuck and aeromancer skippy targeting don't seem to occur..

aika092 commented on Trap Quest.inform/Inform/Extensions/Furniture/Hotel Bed.i7x:
```diff
@@ -15,7 +15,15 @@ To compute furniture resting on (F - a hotel bed):
 	say "You [one of]get into the bed and [or]continue to [stopping]relax on the comfortable mattress.".
 
-HotelBedBoring is a kind of hotel bed.  There are 2 HotelBedBoring.
+HotelBedHalfPosted is a kind of hotel bed. The description is "A bed with four posts. Two high headboard posts and two footboard posts[if the sex addiction of the player > 9] exactly the right height[end if].".
```
The idea with the hotel beds is that until they are slept in for the first time they all have the same description when they are examined, so that the player doesn't know what effect they are about to trigger.  Replacing the boring hotel beds with these breaks that mechanic, as now there's a way to determine which bed is which without sleeping on it.  I'm not sure I'm happy about that.

picobyte replied:
Ah, I see. We can then either extend it each of the 4 hotel beds, save it for a different kind of bed, or just leave it the way it was.

picobyte commented:
I have split this in separate branches, which I hope will make it easier to merge.

topless_addiction: up to 36419 + cherry-pick of commit 55353
some_janitoring:  up to 409b0 (== topless_addiction) + cherry-picks 36419 and c03232
(61cbfe cannot yet be added as it contains a bug, fixed in 96703)

(next will be the bedpost commit with d6de33 and 213e6f)
..and finally the skippy ball updates.
